selfdrive/modeld/modeld.cc

```diff
@@ -80,20 +65,22 @@ void run_model(ModelState &model, VisionIpcClient &vipc_client) {
   double last = 0;
   uint32_t run_count = 0;
 
+  mat3 model_transform = {};
+  bool live_calib_seen = false;
+
   while (!do_exit) {
     VisionIpcBufExtra extra = {};
     VisionBuf *buf = vipc_client.recv(&extra);
     if (buf == nullptr) continue;
 
-    transform_lock.lock();
-    mat3 model_transform = cur_transform;
-    bool valid = live_calib_seen;
-    transform_lock.unlock();
-
     // TODO: path planner timeout?
     sm.update(0);
     int desire = ((int)sm["lateralPlan"].getLateralPlan().getDesire());
     frame_id = sm["roadCameraState"].getRoadCameraState().getFrameId();
+    if (sm.updated("liveCalibration")) {
+      model_transform = update_calibration(sm["liveCalibration"].getLiveCalibration(), wide_camera);
+      live_calib_seen = true;
```

Comment:
use sm.rcv_frame instead?

Reply:
We still need a variable to save the result of sm.rcv_frame and pass it to 2 functions.the current implementation may be more efficient which can reduce a map lookup in sm.rcv_frame